### PR TITLE
fix(query-utils): Escape double quotes from query

### LINKED ISSUE #149: error when importing product with sku containing double quotes
When trying to import following CSV file:
```
productType,sku,variantId 
ptype,"eqsmlg-9'2""",1
```
the import fails with following error:

> BadRequest: Malformed parameter: where: Syntax error while parsing 'where'. Invalid input '"', expected comma or ')' (line 1, column 35):
masterVariant(sku in ("eqsmlg-9'2"")) or variants(sku in ("eqsmlg-9'2""))
                                  ^
    at /app/node_modules/sphere-node-sdk/lib/services/base.js:501:22
    at ProductProjectionService.BaseService._wrapResponse (/app/node_modules/sphere-node-sdk/lib/services/base.js:516:11)
    at /app/node_modules/sphere-node-sdk/lib/services/base.js:358:40
    at Request._callback (/app/node_modules/sphere-node-sdk/lib/connect/rest.js:164:14)
    at Request.self.callback (/app/node_modules/sphere-node-sdk/node_modules/request/request.js:360:22)
    at emitTwo (events.js:87:13)
    at Request.emit (events.js:172:7)
    at Request.<anonymous> (/app/node_modules/sphere-node-sdk/node_modules/request/request.js:1202:14)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:169:7)
    at IncomingMessage.<anonymous> (/app/node_modules/sphere-node-sdk/node_modules/request/request.js:1150:12)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:921:12)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)

### PR DESCRIPTION
**Pull request by wizzy25**
fixes #149

### CODE REVIEW

File: src/coffee/queryutils.coffee [hunk 1]
```diff
@@ -63,12 +63,18 @@ class QueryUtils
...
   @formatCustomAttributePredicate: (name, items) ->
-    "attributes(name=\"#{name}\" and value in (\"#{items.join('", "')}\"))"
+    "attributes(name=\"#{name}\" and value in (#{@escapeItems items}))"
 
   @applyPredicateToVariants: (predicate) ->
     "masterVariant(#{predicate}) or variants(#{predicate})"
 
+  @escapeItems: (items) ->
+    if items.length is 0
+      "\"\""
```
Comment (hisabimbola) [inline src/coffee/queryutils.coffee]: The escape is unnecessarily.

`'""'`

File: src/coffee/queryutils.coffee [hunk 2]
```diff
@@ -63,12 +63,18 @@ class QueryUtils
...
+    "#{name} in (#{@escapeItems items})"
 
   @formatCustomAttributePredicate: (name, items) ->
-    "attributes(name=\"#{name}\" and value in (\"#{items.join('", "')}\"))"
+    "attributes(name=\"#{name}\" and value in (#{@escapeItems items}))"
 
   @applyPredicateToVariants: (predicate) ->
     "masterVariant(#{predicate}) or variants(#{predicate})"
 
+  @escapeItems: (items) ->
```
Comment (hisabimbola) [inline src/coffee/queryutils.coffee]: You can add a default `[]`

File: src/spec/queryutils.spec.coffee [hunk 3]
```diff
@@ -120,15 +120,15 @@ describe "QueryUtils", ->
     it "should return query predicte based on products provided", ->
       predicate = QueryUtils.matchBySku(products)
       expect(predicate).toEqual("masterVariant(sku in " +
-      "(\"1\", \"4\")) or " +
-      "variants(sku in (\"1\", \"4\"))")
+      "(\"1\\\"foo\\\"\", \"4\")) or " +
```
Comment (hisabimbola) [inline src/spec/queryutils.spec.coffee]: why are there triple backslash

Comment (wizzy25) [inline src/spec/queryutils.spec.coffee]: Escaping the backslash that gets compared to the escaped string. I can replace these with single quotes to increase readability but as far as conventions go, there's almost no single quote in this file.

File: src/coffee/queryutils.coffee (hunk 1 above)
Comment (wizzy25) [inline src/coffee/queryutils.coffee]: See [here](https://github.com/sphereio/sphere-node-product-csv-sync/pull/164#discussion_r105125288)

File: src/coffee/queryutils.coffee (hunk 2 above)
Comment (wizzy25) [inline src/coffee/queryutils.coffee]: Why? No other function has defaults

**Comment by wizzy25** [comment]
Wouldn't it be better to use a default in all functions or none than to use default in only one function and leave out the rest? @hisabimbola

File: src/coffee/queryutils.coffee (hunk 2 above)
Comment (hisabimbola) [inline src/coffee/queryutils.coffee]: We can do without it, not necessarily.

Comment (wizzy25) [inline src/coffee/queryutils.coffee]: LGTM

**Pull request merged**